### PR TITLE
fix(mobile): ios random logout

### LINKED ISSUE #8146: Immich iOS keeps logging out after a day or so
### The bug

I recently switched from exposing a port on my network to using Cloudflare tunnels to access Immich over the web. Works great, however I’ve noticed that I keep getting logged out after some time. 

Sometimes it’s a day, other times a couple. Seems to be when I’m off local internet for some time. I read that some network resiliency was added a while back but not sure if it’s entirely functional. Cloudflare tunnel doesn’t report to be unreachable at any given time.

### The OS that Immich Server is running on

unRAID 6.12 - Docker

### Version of Immich Server

1.98.1

### Version of Immich Mobile App

1.99.0 b 145

### Platform with the issue

- [ ] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
version: "3.8"

services:
  immich-server:
    container_name: immich_server
    image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
    command: [ "start.sh", "immich" ]
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
    ports:
      - 2283:3001
    env_file:
      - .env
    depends_on:
      - redis
      - database
    restart: always

  immich-microservices:
    container_name: immich_microservices
    image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
    # extends:
    #   file: hwaccel.yml
    #   service: hwaccel
    command: [ "start.sh", "microservices" ]
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
    env_file:
      - .env
    depends_on:
      - redis
      - database
    restart: always

  immich-machine-learning:
    container_name: immich_machine_learning
    image: ghcr.io/immich-app/immich-machine-learning:${IMMICH_VERSION:-release}
    volumes:
      - model-cache:/cache
    env_file:
      - .env
    restart: always

  redis:
    container_name: immich_redis
    image: redis:6.2-alpine@sha256:70a7a5b641117670beae0d80658430853896b5ef269ccf00d1827427e3263fa3
    restart: always

  database:
    container_name: immich_postgres
    image: registry.hub.docker.com/tensorchord/pgvecto-rs:pg14-v0.2.0@sha256:90724186f0a3517cf6914295b5ab410db9ce23190a2d9d0b9dd6463e3fa298f0
    env_file:
      - .env
    environment:
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_USER: ${DB_USERNAME}
      POSTGRES_DB: ${DB_DATABASE_NAME}
    volumes:
      - pgdata:/var/lib/postgresql/data
    restart: always

volumes:
  pgdata:
  model-cache:
  tsdata:
```

### Your .env content

```Shell
# You can find documentation for all the supported env variables at https://immich.app/docs/install/environment-variables

# The location where your uploaded files are stored
UPLOAD_LOCATION=./immich-data

# The Immich version to use. You can pin this to a specific version like "v1.71.0"
IMMICH_VERSION=release

# Connection secrets for postgres and typesense. You should change these to random passwords
TYPESENSE_API_KEY=key
DB_PASSWORD=passwordb

# The values below this line do not need to be changed
###################################################################################
DB_HOSTNAME=immich_postgres
DB_USERNAME=username
DB_DATABASE_NAME=dbname

REDIS_HOSTNAME=immich_redis
```

### Reproduction steps

```bash
1. Login at home
2. Works fine for some time
3. Come back randomly a few days later and I’m logged out. Rest of family also is
...
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by alextran1502**
fix #8146

The issue is in the method `_isEndpointAvailable`

```diff
  Future<bool> _isEndpointAvailable(String serverUrl) async {
    final Client client = Client();

    if (!serverUrl.endsWith('/api')) {
      serverUrl += '/api';
    }

    try {
      final response = await client
          .get(
            Uri.parse(serverUrl),
          )
          .timeout(const Duration(seconds: 5));

+      if (response.statusCode == 502) {
+        _log.severe(
+          "Server Gateway Error: ${response.body} - Cannot communicate to the server",
+        );
+        return false;
+      }
    } on TimeoutException catch (_) {
      return false;
    } on SocketException catch (_) {
      return false;
    }
    return true;
  }
```

Even when the server is down, the get request can return a response with HTTP status code 502, which should be checked and handled similarly to TimoutException or SocketException to enter the offline mode.

### CODE REVIEW

File: mobile/lib/modules/login/providers/authentication.provider.dart [hunk 1]
```diff
@@ -181,10 +181,21 @@ class AuthenticationNotifier extends StateNotifier<AuthenticationState> {
       UserResponseDto? userResponseDto;
       try {
         userResponseDto = await _apiService.userApi.getMyUserInfo();
-      } on ApiException catch (e) {
-        if (e.innerException is SocketException) {
+      } on ApiException catch (error, stackTrace) {
+        _log.severe(
+          "Error getting user information from the server [API EXCEPTIOON]",
```
Comment (mmomjian) [inline mobile/lib/modules/login/providers/authentication.provider.dart:186]: “EXCEPTION”?

**Review by mmomjian — APPROVED** [review]
LGTM. Hopefully at a minimum the increased logging will help provide info. 

Do we know why the 502 is getting hit / server is unreachable? From my external monitoring tools (which checks for both page content and HTTP return code) Immich Web and API endpoints are rock solid.

File: mobile/lib/shared/services/api.service.dart [hunk 2]
```diff
@@ -95,14 +97,19 @@ class ApiService {
...
-    // we do not care if the endpoints hits an HTTP error
     try {
-      await client
+      final response = await client
           .get(
             Uri.parse(serverUrl),
           )
           .timeout(const Duration(seconds: 5));
+
+      if (response.statusCode == 502) {
```
Comment (jrasm91) [inline mobile/lib/shared/services/api.service.dart:107]: You should probably do the opposite actually. If it isn't a 200, log the error and return false.

**Review by jrasm91 — CHANGES_REQUESTED** [review]

File: mobile/lib/shared/services/api.service.dart (hunk 2 above)
Comment (alextran1502) [reply mobile/lib/shared/services/api.service.dart]: Done; it never hit a valid GET endpoint, so it returned 404 or 512. I modify to use GET /server-info/ping to get a valid HTTP 200 when possible

Comment (mmomjian) [reply mobile/lib/shared/services/api.service.dart]: Will $serverUrl contain the /api suffix even if not provided on the login screen? I know it works without it currently but im not sure where that gets added/checked.

**Review by jrasm91 — APPROVED** [review]

File: mobile/lib/shared/services/api.service.dart (hunk 2 above)
Comment (alextran1502) [reply mobile/lib/shared/services/api.service.dart]: @mmomjian it gets added if it is not included

**Pull request merged**